src/js/actions/menu.js

```diff
@@ -63,18 +74,38 @@ define(function (require, exports) {
             return Promise.reject(error);
         }
 
-        // Photoshop expects commandId with a lower case d, so convert here
-        payload.commandId = payload.commandID;
-
-        return ps.performMenuCommand(payload).then(function (success) {
-            // FIXME: After the M2 release, remove the debug conjunct
-            if (global.debug && !success) {
-                throw new Error("Menu command not available: " + payload.commandID);
-            }
-        });
+        var isPlaceCommand = payload.commandID === _PLACE_LINKED_MENU_ID ||
+                payload.commandID === _PLACE_EMBEDDED_MENU_ID;
+        
+        return Promise.bind(this)
+            .then(function () {
+                if (isPlaceCommand) {
+                    this.dispatchAsync(events.menus.PLACE_COMMAND, { executing: true });
```

Comment:
You probably want to just use `this.dispatch` here? Otherwise, the dispatch may not be complete by the time your transfer command executes.